Tweak Gain arithmetic (#463)
PainterQubits/Unitful.jl: docs/src/logarithm.md

20 mW
```

With a few exceptions, dimensionful logarithmic units, such as `dBm` behave just like the underlying

> linear unit for purposes of arithmetic (i.e. arithmetic operations commute with `linear`). However,

Reviewer: `linear` is not yet introduced at this point, so users will not know what this means.

    Unitful.Gain
```

> One might expect that any gain / attenuation factor should be convertible to a scalar,

Reviewer: I agree that “pure number” is not a good term here, but I also don’t like “scalar”. To me, scalar means “not a vector”, and even though that is technically correct (dimensionful quantities form a vector space), I think it doesn’t make this more clear. I would write “real number”, but I see that you replaced that as well. Can you explain why you prefer “scalar” over “real number”?

If we agree on “scalar” here, it should also be used in the first sentence in the “Constructing logarithmic quantities” section.

    Unitful.LogScaled
```

> ## Addition and Multiplication rules

Reviewer: ```suggestion
## Addition and multiplication rules
```

> For dimensionaless logarithmic units, addition behaves as one might expect:

Reviewer: ```suggestion
For dimensionless logarithmic quantities, addition behaves as one might expect:
```

```jldoctest
julia> 10u"dB" + 10u"dB"
20 dB
```

> I.e. the gains add. However, as hinted at above, dimensionful logarithmic units,
> behave as their corresponding linear unit:

Reviewer: ```suggestion
I.e. the gains add. However, as hinted at above, dimensionful logarithmic quantities
behave as their corresponding linear quantity:
```

```
julia> 10u"dBm" + 10u"dBm"
13.010299956639813 dBm

julia> linear(10u"dBm") + linear(10u"dBm")
20.0 mW

julia> uconvert(u"dBm", ans)
13.010299956639813 dBm
```

Note that this may seem strange from an arithmetic perspective, as written, but
the arithmetic is entirely consistent. It can be helpful to think of the arithmetic
as being performed on the linear units, with the logarithmic units simply being a
display hint (although the quantity being stored is indeed the displayed logarithmic

> value).

Reviewer: This is not true. `Level` does actually store the linear value:
```julia
julia> @dB 100u"mW"/1u"mW"
20.0 dBm

julia> ans.val
100 mW
```

[0 dB] Hz^-1
```

> Since dimensionful logarithmic units still behave as their corresponding linear unit,

Reviewer: ```suggestion
Since dimensionful logarithmic quantities still behave as their corresponding linear quantities,
```

> working with dimensionful units is entirely consistnet.

Reviewer: ```suggestion
working with dimensionful units is entirely consistent.
```

> The behavior of addition and multiplication are summarized in the following table, with entries marked by

Reviewer: ```suggestion
The behavior of addition and multiplication is summarized in the following tables, with entries marked by
```

10.0 V
```

## Mixed Arithmetic

One final question to answer is how arithmetic behaves when it involves both dimensionless
and dimensionful logarithmic units. The answer here is that in mixed arithmetic, both
dimensionless and dimensionful units are treated logarithmically. This is done for
convenience and can break commutativity and associativity, so should be probably avoided in generic
code.

```
julia> 10u"dBm" + 20u"dB"
30.0 dBm

julia> (10u"dBm" + 10u"dBm") + 20u"dB"
33.01029995663981 dBm

julia> 10u"dBm" + (10u"dBm" + 20u"dB")
30.043213737826427 dBm

julia> 10u"dBm" * 20u"dB"
ERROR: ArgumentError: Multiplying a level by a Gain is disallowed. Use addition, or `linear` depending on context.

> julia> linear(10u"dBm") * 20u"dB"

Reviewer: ```suggestion
julia> 10u"mW" * 20u"dB"
```
I think this would make it more clear.

1000.0 mW

> julia> linear(10u"dBm") + 20u"dB"

Reviewer: ```suggestion
julia> 10u"mW" + 20u"dB"
```
